### ISSUE TITLE
Can't use XML element names whose first character is not ASCII in queries

### ISSUE THREAD

**Issue by nanis** [manwar/XML-XPath#4]
First reported [here]. I subsequently replicated the problem with this short script:

```perl
#!/usr/bin/env perl

use v5.14;
use strict;
use warnings;

use utf8;
use open qw(:std :encoding(UTF-8));

use XML::XPath;

my $xp = XML::XPath->new(ioref => \*DATA );

my $good_path = '/employees/employee[@age="30"]/yağcı';
my $bad_path = '/employees/employee[@age="30"]/şımarık';

say $xp->findvalue($good_path);

say $xp->findvalue($bad_path);

__DATA__
<?xml version="1.0" encoding="utf-8" ?>
<employees>
    <employee age="30">
        <şımarık>değil</şımarık>
        <yağcı>değil</yağcı>
    </employee>
</employees>
```

The problem seems to be caused by [this pattern on line 45 of XML/XPath/Parser.pm][problem]:

    $NCName = '([A-Za-z_][\w\\.\\-]*)';

This pattern restricts the first character, and only the first character, of the name of an element in a query to the letters of the English alphabet and `_`.

I am not sure if there is a deep reason for the restriction, but it seems unlikely that there would be. Changing that pattern to 

    $NCName = '(\w[\w\\.\\-]*)';

solves the problem for the case above.

Of course, the solution is to use the correct definition from the [standard](https://www.w3.org/TR/REC-xml/#NT-Name).

Thank you for your help.

[here]: https://stackoverflow.com/questions/45307294/how-to-work-with-xmlxpath-in-utf8
[problem]: https://github.com/manwar/XML-XPath/blob/d39003bd2306d0c4296ae5dacb5816ea3d0fe610/lib/XML/XPath/Parser.pm#L45

**Comment by nanis** [comment]
As [@ikegami][ikegami] points out in the above mentioned Stackoverflow thread, the correct specification for the first character is [given in the standard][standard]:

```
NameStartChar	   ::=   	":" | [A-Z] | "_" | [a-z] | [#xC0-#xD6] | [#xD8-#xF6] | [#xF8-#x2FF] | [#x370-#x37D] | [#x37F-#x1FFF] | [#x200C-#x200D] | [#x2070-#x218F] | [#x2C00-#x2FEF] | [#x3001-#xD7FF] | [#xF900-#xFDCF] | [#xFDF0-#xFFFD] | [#x10000-#xEFFFF]
```

[ikegami]: https://stackoverflow.com/users/589924/ikegami
[standard]: https://www.w3.org/TR/REC-xml/#NT-Name

**Comment by manwar** [comment]
Thanks @nanis for reporting the issue.
I will try to implement the standard as you mentioned above.

Best Regards,
Mohammad S Anwar

**Comment by nanis** [comment]
See also [@ikegami's answer][1] which contains code you should be able to use with attribution.

[1]: https://stackoverflow.com/a/45307470/100754

**Comment by manwar** [comment]
XML::XPath v1.41 has been released to path the issue #4.

https://metacpan.org/release/MANWAR/XML-XPath-1.41
